src/graphql/search.rs

```diff
@@ -0,0 +1,294 @@
+                    status_msg: item.status_msg.clone(),
+                    username: item.username.clone(),
+                    password: item.password.clone(),
+                    cookie: item.cookie.clone(),
+                    content_encoding: item.content_encoding.clone(),
+                    content_type: item.content_type.clone(),
+                    cache_control: item.cache_control.clone(),
+                };
+                if let Some(filter) = filter.as_ref() {
+                    match check(filter, &item, &k.timestamp, &k.name) {
+                        Ok(true) => records.push(raw_event.clone()),
+                        Ok(false) | Err(_) => {}
+                    }
+                }
+                copy.push(raw_event);
+            }
+        }
+
+        match filter {
+            Some(_) => Ok(records),
+            None => Ok(copy),
+        }
+    }
+}
+
+#[allow(clippy::too_many_lines)]
+fn check(
+    filter: &SearchFilter,
+    item: &Http,
+    timestamp: &DateTime<Utc>,
+    name: &String,
+) -> Result<bool> {
+    if let Some(keyword) = &filter.keyword {
+        let matcher = AhoCorasick::new([keyword.clone()]);
+        if matcher.is_match(name)
+            || matcher.is_match(item.orig_addr.to_string())
+            || matcher.is_match(item.orig_port.to_string())
+            || matcher.is_match(item.resp_addr.to_string())
+            || matcher.is_match(item.resp_port.to_string())
+            || matcher.is_match(item.proto.to_string())
+            || matcher.is_match(item.last_time.to_string())
+            || matcher.is_match(&item.method)
+            || matcher.is_match(&item.host)
+            || matcher.is_match(&item.uri)
+            || matcher.is_match(&item.referrer)
+            || matcher.is_match(&item.version)
+            || matcher.is_match(&item.user_agent)
+            || matcher.is_match(item.request_len.to_string())
+            || matcher.is_match(item.response_len.to_string())
+            || matcher.is_match(item.status_code.to_string())
+            || matcher.is_match(&item.status_msg)
+            || matcher.is_match(&item.username)
+            || matcher.is_match(&item.password)
+            || matcher.is_match(&item.cookie)
+            || matcher.is_match(&item.content_encoding)
+            || matcher.is_match(&item.content_type)
+            || matcher.is_match(&item.cache_control)
+        {
+            return Ok(true);
+        }
+        return Ok(false);
+    }
+
+    if let Some(src_ip_range) = &filter.src_ip {
+        let start = if let Some(start) = &src_ip_range.start {
+            item.orig_addr >= start.parse::<IpAddr>()?
+        } else {
+            false
+        };
+
+        let end = if let Some(end) = &src_ip_range.end {
+            item.orig_addr <= end.parse::<IpAddr>()?
+        } else {
+            false
+        };
+
+        if end || start {
+            return Ok(false);
+        };
+    }
```

Comment:
I think the following would be clearer

```
if let Some(src_ip_range) = &filter.src_ip {
        let (start, end) =
            if let (Some(start), Some(end)) = (&src_ip_range.start, &src_ip_range.end) {
                (start.parse::<IpAddr>()?, end.parse::<IpAddr>()?)
            } else {
                return Ok(false);
            };

        if item.orig_addr >= start || item.orig_addr <= end {
            return Ok(false);
        }
    }
```

And, I think this part can be a function, so that the below redundant codes will be simpler.